File: Taller de Haskell/Python/problem8.py
# Precios de Arcade

# Criterio:
# <7 años → gratis
# 7–18 años → $18000
# >18 años → $28000

edad = int(input("Ingrese su edad: "))

# Condicionales
if edad < 0 or edad > 100:
    print("Edad no válida.")
elif edad < 7:
    boleta = 0
    print("Entrada gratis.")
elif edad <= 18:
    boleta = 18000
    print(f"El precio de la entrada es ${boleta} ")
else:
    boleta = 28000
    print(f"El precio de la entrada es ${boleta} ")

File: Taller de Haskell/Haskell/problem8.hs
-- Precios de Arcade
-- <7 años → gratis
-- 7–18 años → $18000
-- >18 años → $28000
-- Validación: edad negativa o >100 = inválida

precioArcade :: Int -> String
precioArcade edad
    | edad < 0 || edad > 100 = "Edad no válida."
    | edad < 7 = "Entrada gratis."
    | edad <= 18 = "El precio de la entrada es $18000"
    | otherwise = "El precio de la entrada es $28000"

main :: IO ()
main = do
    putStrLn "Ingrese su edad: "
    input <- getLine
    let edad = read input :: Int
    putStrLn (precioArcade edad)